### ISSUE TITLE
ActiveWindowName won't respect max_chars

### ISSUE THREAD

**Issue by greweln** [sminez/penrose#286]
### Describe the bug
The widget displays full length text ignoring the max_chars value.
...

### To Reproduce
- On the same workspace,  st + firefox (with some tabs).  

- While keeping the focus on the Firefox window switch tabs.It will display the full URL ignoring the max_chars value.  If the  URL is long enough will even "push " the next widgets out of the screen.

- Switching the focus between  st/firefox(while having the tab with the long URL displayed) works as expected.
### Expected behavior

...

### Screenshots
![screenshot](https://github.com/sminez/penrose/assets/78338967/f5a1cab1-5ce3-47b6-8a7b-caeca651f09c)

**Comment by sminez** [comment]
Hey @greweln, sorry about the delay on this one! I think the fix might be embarrassingly simple...

https://github.com/sminez/penrose/blob/develop/crates/penrose_ui/src/bar/widgets/simple.rs#L129

Pretty sure that all that is required is to change that line from
```rust
self.inner.set_text(x.window_title(*id)?)
```
to
```rust
self.set_text(x.window_title(*id)?)
```

I've just pushed that change on a branch (`issue-286`) are you able to try that out and verify if it fixes the issue for you?

**Comment by greweln** [comment]
@sminez 
I think it should be borrowed here: 
 `self.set_text(&x.window_title(*id)?)`
right?

**Comment by sminez** [comment]
The fact that those two signatures don't match makes me all kinds of sad 😞

And yeah, that should either be a reference or the signature should be updated so it matches Text::set_text

**Comment by sminez** [comment]
fixed in #287
